pkg/querier/worker/scheduler_processor_test.go

```diff
@@ -101,6 +102,45 @@ func TestSchedulerProcessor_processQueriesOnSingleStream(t *testing.T) {
 		loopClient.AssertCalled(t, "Send", &schedulerpb.QuerierToScheduler{QuerierID: "test-querier-id"})
 	})
 
+	t.Run("should pass the query tags from the request in the context", func(t *testing.T) {
```

Comment:
I think this is testing the internals, not the desired behaviour.

What we want is for the query tags to be received in the querier when sent by the frontend; whether it's passed in the context or not is an implementation detail.

IMO this test should be instantiating an HTTP request, passing it to the frontend, and the querier worker should receive it with all the required parameters present.

Reply:
But how would you test that the querier worker receives it? The only place it's applied is on the logs as far as I can tell.